contracts/asset-proxy/contracts/src/ERC1155Proxy.sol

```diff
@@ -0,0 +1,268 @@
+                // been constructed in memory, the destination erc1155 contract is called using this 
+                // as its calldata. This process is divided into four steps, below.
+
+                ////////// STEP 1/4 //////////
+                // Map relevant fields from assetData (Table #2) into memory (Table #3)
+                // The Contents column of Table #2 is the same as Table #3,
+                // beginning from parameter 3 - `offset to tokenIds (*)`
+                // The offsets in these rows are offset by 32 bytes in Table #3.
+                // Strategy:
+                // 1. Copy the assetData into memory at offset 32
+                // 2. Increment by 32 the offsets to `tokenIds`, `tokenValues`, and `callbackData`
+
+                // Load offset to `assetData`
+                let assetDataOffset := calldataload(4)
+
+                // Load length in bytes of `assetData`, computed by:
+                // 4 (function selector)
+                // + assetDataOffset
+                let assetDataLength := calldataload(add(4, assetDataOffset))
+
+                // This corresponds to the beginning of the Data Area for Table #3.
+                // Computed by:
+                // 4 (function selector)
+                // + assetDataOffset
+                // + 32 (length of assetData)
+                calldatacopy(32, add(36, assetDataOffset), assetDataLength)
+
+                // Increment by 32 the offsets to `tokenIds`, `tokenValues`, and `callbackData` 
+                mstore(68, add(mload(68), 32))
+                mstore(100, add(mload(100), 32))
+                mstore(132, add(mload(132), 32))
+
+                // Record the address of the destination erc1155 asset for later.
+                let assetAddress := and(
+                    mload(36),
+                    0x000000000000000000000000ffffffffffffffffffffffffffffffffffffffff
+                )
+
+                ////////// STEP 2/4 //////////
+                let scaleAmount := calldataload(100)
+                let tokenValuesOffset := add(mload(100), 4) // add 4 for calldata offset
+                let tokenValuesLengthInBytes := mul(
+                    mload(tokenValuesOffset),
+                    32
+                )
+                let tokenValuesBegin := add(tokenValuesOffset, 32)
+                let tokenValuesEnd := add(tokenValuesBegin, tokenValuesLengthInBytes)
+                for { let tokenValueOffset := tokenValuesBegin }
+                    lt(tokenValueOffset, tokenValuesEnd)
+                    { tokenValueOffset := add(tokenValueOffset, 32) }
+                {
+                    // Load token value and generate scaled value
+                    let tokenValue := mload(tokenValueOffset)
+                    let scaledTokenValue := mul(tokenValue, scaleAmount)
+                    
+                    // Check if scaled value is zero
+                    if iszero(scaledTokenValue) {
```

Comment:
We don't actually check if the `amount` is 0 in any of the other proxies. What is the reason for doing so here, and can this ever be potentially limiting?